### PR TITLE
Schemas: Add plumbing for `__prefect_kind` and implement `json`

### PR DESCRIPTION
**Pull request by pleek91**
# Description
Adds logic and components for switching between different types of inputs and adds the ability to switch between using a form or using json at any level. 

image

image

### CODE REVIEW

File: src/schemas/components/SchemaFormKindInput.vue
Comment (pleek91) [inline src/schemas/components/SchemaFormKindInput.vue:1]: Also including the switch from `null` to `"none"` for the noop kind.

File: src/schemas/components/SchemaFormPropertyAnyOf.vue
Comment (pleek91) [inline src/schemas/components/SchemaFormPropertyAnyOf.vue:1]: Nothing changed functionally in this file. I was referencing it a few times and organized it a bit to make it more readable.

File: src/schemas/components/SchemaFormPropertyInput.vue [hunk 1]
```diff
@@ -93,6 +93,8 @@
     }
 
     if (isSchemaPropertyType(type, undefined)) {
+      // automatically switch to prefect_kind === 'json'?
+      // value might not be json, so will need to convert it to json...
       throw 'not implemented'
```
Comment (znicholasbrown) [inline src/schemas/components/SchemaFormPropertyInput.vue:98]: In what scenarios would we need to convert the value? If it's coming from the db it should already be serialized so we shouldn't need do anything here I _think_

**Review by znicholasbrown — APPROVED** [review]
1 non-blocking comment, otherwise this looks like i'd expect

**Pull request merged**